dart/lib/src/sentry_options.dart

```diff
@@ -20,45 +20,32 @@ class SentryOptions {
   /// Default Log level if not specified Default is DEBUG
   static final SentryLevel _defaultDiagnosticLevel = SentryLevel.debug;
 
-  /// The DSN tells the SDK where to send the events to. If this value is not provided, the SDK will
-  ///  just not send any events.
-  String dsn;
-
-  bool _compressPayload = true;
+  /// The DSN tells the SDK where to send the events to. If an empty string is
+  /// used, the SDK will not send any events.
+  String dsn = '';
```

Comment:
empty DSN means SDK is disabled, no DSN at all, means, you have a problem, I'd make it nullable